dubbo-common/src/main/java/org/apache/dubbo/common/utils/CompatibleTypeUtils.java

```diff
@@ -129,6 +129,8 @@ public static Object compatibleTypeConvert(Object value, Class<?> type) {
                 return BigDecimal.valueOf(number.doubleValue());
             } else if (type == Date.class) {
                 return new Date(number.longValue());
+            } else if (type == boolean.class || type == Boolean.class) {
+                return 1 == number.intValue() ? Boolean.TRUE : Boolean.FALSE;

```

Comment:
please change to `return number.intValue() != 0`:
0 return false, else return true.

Reply:
It has been modified